activesupport/test/core_ext/duration_test.rb

```diff
@@ -111,7 +111,11 @@ def test_multiply
   def test_divide
     assert_equal 1.day, 7.days / 7
     assert_instance_of ActiveSupport::Duration, 7.days / 7
-    assert_equal 1, 1.day / 1.day
+    assert_equal 1.day, 1.day / 1.day
+
+    assert_equal 2.days, 4.days / 2.days
+    assert_equal 2.days, 4.days / 1 / 2.days
+    assert_instance_of ActiveSupport::Duration, 4.days / 1 / 2.days
```

Comment:
Mathematically speaking, the units should cancel out. I'd much rather prefer the following behavior, which was the case prior to 5.1.0:

```
assert_equal 1.day, 7.days / 7
assert_instance_of ActiveSupport::Duration, 7.days / 7

assert_equal 1, 1.day / 1.day
assert_instance_of Numeric, 1.day / 1.day

assert_equal 90.0, 1.5.hours / 1.minute
assert_instance_of Numeric, 1.5.hours / 1.minute
```